src/nlua_misn.c

```diff
@@ -1020,3 +1029,95 @@ static int misn_claim( lua_State *L )
 
 
 
+
+/**
+ * @brief Creates a shiplog for this mission.
+ *
+ *    @luatparam string Name for this log.
+ *    @luatparam string Type of log (e.g travel, trade, etc, can be anything.)
+ *    @luatparam number Whether to overwrite existing mission with this logname and logtype.  Warning, removes previous entries of this logname and type.
+ *    @luatreturn number The logid of the mission.
+ * @luafunc createLog( logname, logtype, overwrite )
+ * @usage misn.createLog("My mission title","Mission type",0)
+ * @usage misn.createLog("Any title","Anything can be a type",1) with 1 to replace existing missions of this title and type.
+ */
+static int misn_createLog( lua_State *L )
+{
+   const char *logname;
+   const char *logtype;
+   int overwrite;
+   Mission *cur_mission;
+   /* Parameters. */
+   logname    = luaL_checkstring(L,1);
+   logtype    = luaL_checkstring(L,2);
+   overwrite = luaL_checkint(L,3);
+
+   cur_mission = misn_getFromLua(L);
+
+   /* Create a new shiplog */
+   cur_mission->logid = shiplog_create( logname, logtype, overwrite );
+
+   lua_pushnumber(L, cur_mission->logid);
+   return 1;
+}
+  
+/**
+ * @brief Appends to the shiplog.
+ *
+ *    @luatparam string message to append to the log.
+ * @luafunc appendLog( message )
+ * @usage misn.appendLog("Some message here")
+ */
+static int misn_appendLog( lua_State *L )
+{
+   const char *msg;
+   Mission *cur_mission;
+   int ret;
+   msg = luaL_checkstring(L, 1);
+   cur_mission = misn_getFromLua(L);
+   if ( cur_mission->logid == -1 ){
+      cur_mission->logid = shiplog_create( "Unnamed mission", "Miscellaneous", 0 );
+   }
+   ret = shiplog_append(cur_mission->logid, msg);
+
+   lua_pushnumber(L, ret); /* 0 on success, -1 on failure */
+   return 1;
+}
+
+
+/**
+ * @brief Deletes the shiplog.
+ *
+ * @luafunc deleteLog( )
+ * @usage misn.deleteLog()
+ */
+static int misn_deleteLog( lua_State *L )
```

Comment:
I think you should remove this Lua function, too. Deletion of logs should always be done by the user if it's done at all. Missions shouldn't be in the habit of making missions disappear automatically.

🕵️